src/markdown.js

```diff
@@ -0,0 +1,60 @@
+const cheerio = require('cheerio');
+const marked = require('marked');
+const shared = require('./utils/shared');
+
+const getRenderer = () => {
+  const renderer = new marked.Renderer();
+
+  renderer.code = (code, language) =>
+`<stache-code-block languageType="${language}">
+${code}
+</stache-code-block>
+`;
+
+  renderer.heading = (text, level) => {
+    let parsedHeading = `
+<h${level}>
+  ${text}
+</h${level}>
+`;
+
+    if (level === 2) {
+      parsedHeading = `
+<stache-page-anchor>
```

Comment:
Hm, do we really want to do this? Currently, a user can write both `<h2>` and `<stache-page-anchor>` separately. If we force all `## xxx` to be page anchors, then the user will never be able to write simple `<h2>`. 

I'd prefer the addition of page anchors to be an explicit action taken by the user, instead of something automatic... What are y'all's thoughts? @Blackbaud-BrandonJones @blackbaud-brandonhare @Blackbaud-BobbyEarl

Reply:
I don't think I agree. In a Stache project, `<h2></h2>` should be page anchors. We have an established pattern there. I think we should make the default behavior be the thing we want them to use. If they want to use a vanilla h2 for some reason, and I think they'd have to have a good reason, then they should be deliberate about it.

Reply:
I think I agree with Hare here. Simply because it's an established pattern from `Stache1`.   The people who want markdown so badly will expect this behavior of an h2.  If they do not want that, then they can wrap it in an `<h2>` tag manually.  I think if you try to include angular `html` inside the markdown file right now it will strip it out as an unknown html element.

Reply:
We have two options here:
Let the `##` render as an `h2`  or as a `page-anchor` like it does in stache 1.   If we let it render as an `h2` then users would have to add a `<stache-page-anchor>` tag manually where they want them.  Or vice versa, they add the `<h2>` when they don't want a page anchor but want an h2.   I feel like people would seldom want to use an `<h2>` without it being a page anchor.  Though perhaps we should ask @Blackbaud-LindseyRix and @blackbaud-johnly their input?

Reply:
We could make that an attribute that's passed to `stache-markdown`?  `renderAsStache` or something like that, which I'd push to have default to `true`.

Reply:
Stache tags render now find with the fix we made to the html stuff :)  
```
<stache-markdown>
<stache-code escapeCharacters="true"><span>demo</span></stache-code>
<ul>
  <li>
    item
  </li>
</ul>

## Title
</stache-markdown>
```

All renders as intended.

Reply:
No, that's not what I'm saying.  I'm saying we could conditionally choose as to whether `##` renders as `<stache-page-anchor>` or as `<h2>`.  I'm all for them rendering as `<stache-page-anchor>`, but could concede that maybe someone wouldn't want that.  In that case, you could turn off all our custom rendering for the plugin.

Also meaning, instead of three back-ticks (couldn't get those to render correctly here) rendering as a `<stache-code-block>` it would be rendered as the default `<code><pre>` block.

Reply:
Hmmm.  So we want to give them the option to write markdown and not use any of the stache components.  Except the `<stache-markdown>` tag and the `<stache>` wrapper itself?  That would kind of be a bit jarring as far as styling consistency across sites, but if we think there is a need/desire for that for some reason I guess I could understand that?

My personal preference (pending on what John and Lindsey have for feedback) would be for the `##` to render as a `stache-page-anchor` and have them use `<h2>` when they don't want to use an anchor.  That way they have to use/remember fewer stache tags, without losing the styling we are going for.

Reply:
Makes sense to me to have ## render as `stache-page-anchor` and have users manually insert `h2` when necessary. ... I'm not opposed to Bobby's idea of an attribute for this, but I suspect that even if make this attribute, people will just use the default and manually insert `h2`s when necessary.

Reply:
If that behavior will be possible, I think that's appropriate (`##` is a page anchor and `<h2>` is preserved).